app/controllers/SubmissionsController.scala

```diff
@@ -73,9 +74,10 @@ class SubmissionsController @Inject()(
     }
   }
 
-  def amendErrors(id: String, redirectUrl: String): Action[AnyContent] = authenticate.async { implicit request =>
+  def amendErrors(id: String, redirectUrl: String, pattern: String, messageKey: String): Action[AnyContent] = authenticate.async { implicit request =>
     val redirectUrlWithMode = redirectUrl + ErrorFix.queryParameter
-    val redirect = Redirect(redirectUrlWithMode)
+    val flashData = Map(errorMessage -> messageKey)
```

Comment:
I'm wondering if you'll be adding the field name to this method.

The error support for the page require error message and field name to display. Here is just error Message. I think it will be good to add it to have it for the future

Reply:
Yes.  When we add the field names, this flashData map will need to be updated.  I believe there is a new ticket to do this.